### ISSUE TITLE
Conflict between validate package and nzchar_linter()

### ISSUE THREAD

**Issue by jabenninghoff** [r-lib/lintr#2790]
`nzchar_linter()` generates a lint for the following code:

``` r
library(validate)
test <- data.frame(name = c("test", ""), stringsAsFactors = FALSE)
validate_nchar <- validator(name_not_blank = nchar(name) > 0)
print(summary(confront(test, validate_nchar)))
```

Output:

```
            name items passes fails nNA error warning      expression
1 name_not_blank     2      1     1   0 FALSE   FALSE nchar(name) > 0
```

However, `nzchar()` is not recognized by `validator()`, which generates a warning (at least when run inside an R Notebook):

``` r
library(validate)
test <- data.frame(name = c("test", ""), stringsAsFactors = FALSE)
validate_nzchar <- validator(name_not_blank = nzchar(name))
print(summary(confront(test, validate_nzchar)))
```

Output:

```
Warning: Invalid syntax detected, the following expressions have been ignored:
[001] nzchar(name)
[1] error   warning
<0 rows> (or 0-length row.names)
```

Is it possible to modify `nzchar_linter()` to ignore `nchar()` expressions within `validator()`? I realize this may be a limited use case, but wanted to ask; I'd appreciate feedback on how best to address this conflict.

**Comment by MichaelChirico** [comment]
I'm not very familiar with this package -- @markvanderloo, any reason why `nzchar()` would be "invalid"?

**Comment by markvanderloo** [comment]
Try `nzchar(name)>0`< like you did with `nchar`

**Comment by jabenninghoff** [comment]
So I also don't see how `nzchar(name)>0` is valid, however, `nzchar(name) == TRUE` does work, full context:

```r
validate_nzchar <- validator(name_not_blank = nzchar(name) == TRUE)
print(summary(confront(test, validate_nzchar)))
```

**Comment by jabenninghoff** [comment]
Update: `nzchar(name) == TRUE` also generates a lint (`redundant_equals_linter()`). `nzchar(name)>0` works (and generates no lints), but seems confusing to me.

**Comment by markvanderloo** [comment]
Validation rules are statically analyzed to decide if they evaluate to a logical. nzchar() is not in the list of functions that evaluates. However, all comparison operators, logical operators and some more are there. See also ?validate::syntax for a brief overview

**Comment by jabenninghoff** [comment]
Is it possible to add `nzchar()` to the list of logical functions? If so, I will close this issue and open one on validate.

**Comment by MichaelChirico** [comment]
I sent a proposal PR to {validate}.

It may be that `redundant_equals_linter()` is just generally not compatible with how {validate} is intended to operate -- you might just disable it in that case.

I don't think there's anything for {lintr} to do here.

**Comment by AshesITR** [comment]
Checking the source code of validate I noticed it also blindly accepts all functions starting with `"is."`, so a temporary workaround could also be `is.nzchar <- nzchar` and then using `is.nzchar` in the validator.
